tt_metal/hw/ckernels/wormhole_b0/common/inc/ckernel_perf_api.h

```diff
@@ -0,0 +1,162 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+#include <l1_address_map.h>
+#include "ckernel_include.h"
+#include "ckernel_globals.h"
+#include "ckernel.h"
+#include "tensix.h"
+#include "fw_debug.h"
+#include "epoch.h"
+
+#ifdef PERF_DUMP
+#include "perf_lib/scratch_api.h"
+#include "perf_res_decouple.h"
+#include "ckernel_perf_math.h"
+#include "ckernel_perf_unpack_pack.h"
+#endif
+
+#ifndef INTERMED_DUMP
+#define INTERMED_DUMP 0
+#endif
+
+#pragma GCC diagnostic ignored "-Wunused-function"
+
+// Comment in/out to enable perf scratch even logging
+
+namespace ckernel
+{
+extern uint32_t perf_index;
+extern uint32_t perf_end;
+// Perf-buffer are double buffered for spill_to_dram.
+// Ncrisc will move one half to dram while trisc populates the other half.
+// When INTERMED_DUMP = 0, we only dump into perf_buf_base[0].
+extern volatile uint32_t *perf_buf_base[2];
+// Selects the half of perf_buffer that trisc is currently writing into.
+extern uint8_t perf_buf_base_id;
+extern bool record_perf_events;
+extern uint32_t perf_events_target_idx;
+extern uint16_t current_outer_loop_iter;
+extern uint8_t thread_id;
+extern bool first_unpack_recorded;
+
+inline void set_perf_dump_flag_for_input(int input_idx) {
+   #ifdef PERF_DUMP
+      TT_LLK_DUMP("set_perf_dump_flag_for_input({})", input_idx);
+      if (perf_events_target_inputs[perf_events_target_idx] == input_idx) {
+         record_perf_events = true;
+         perf_events_target_idx++;
+         if constexpr(PERF_DUMP_CONCURRENT == 0 && INTERMED_DUMP == 0) {
+            if (thread_id == 0 || thread_id == 2) {
+                  perf_end += num_events_per_input;
+                  // The buffer size available for each thread after double buffering is (l1_mem::address_map::TRISC_PERF_BUF_SIZE)/2.
+                  // Max number of events we can record in each half of the buffer will be that size divided by 4, since each event will be 4 bytes.
+                  if (perf_end > (TRISC_PERF_BUF_SIZE >> 2)) {
+                     perf_end = TRISC_PERF_BUF_SIZE >> 2;
+                  }
+            }
+         }
+         current_outer_loop_iter = input_idx;
+      } else {
+         record_perf_events = false;
+      }
+      first_unpack_recorded = false;
+   #endif
```

Comment:
@davorchap Can we remove all this because we use a different perf-profiling mechanism?